ui/hooks/useTransactionEventFragment.js

```diff
@@ -0,0 +1,31 @@
+import { useCallback } from 'react';
+import { TRANSACTION_EVENTS } from '../../shared/constants/transaction';
+
+import { useGasFeeContext } from '../contexts/gasFee';
+import { trackTransactionMetricsEvent } from '../store/actions';
+
+export const useTransactionEventFragment = () => {
```

Comment:
[REQ] I think this function should ultimately have the same signature as the `useEventFragment` method but with the extra bit of using a known id (everything a user can modify in a normal flow will modify the transaction-added fragment, the only one we might want to modify the transaction-submitted fragment is on the speed up or cancel flows). We also need to use the createTransactionEventFragment to ensure the proper fragment exists before we modify it.

Reply:
As mentioned above I would prefer to re-use controller methods `trackTransactionMetricsEvent` for UI, thus keeping things simple at UI for transaction events.

Reply:
using the fragment system will ultimately result in this method being called, that's the issue I'm seeing ... and this method was never meant or intended to be called from the UI.